src/core/Directus/Database/TableGateway/BaseTableGateway.php

```diff
@@ -433,7 +434,7 @@ public function updateRecordByArray(array $recordData)
         $recordId = $recordData[$primaryKey];
         if ($collectionName === SchemaManager::COLLECTION_FILES) {
```

Comment:
Same as above, lets move this `/asset` endpoint specific code out of the deep BaseTableGateway abstraction layer.

Reply:
I would say the same for this too - this is an [older code](https://github.com/directus/api/blob/8d4473c7a1eb1129b24174d5307f845fafe473e3/src/core/Directus/Database/TableGateway/BaseTableGateway.php). Let me know - if we are supposed to clean it in this PR.